### PR TITLE
feat(playground): Sync rome config to local storage

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
## Summary
1. Resolve https://github.com/rome/tools/issues/2846

## Test Plan
1. Check out if the new feature of the playground is expected.

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
https://user-images.githubusercontent.com/17974631/178101372-68fdc7ac-b07a-4b02-84d9-91555963164f.mp4

File: website/playground/src/utils.ts [hunk 1]
```diff
@@ -194,3 +217,73 @@ function fromBinary(binary: string) {
...
+	return mergeRomeConfig(config, defaultRomeConfig);
+}
+
+// Change the romeConfig in local storage seems to be meaningless, so we simplify the validation phase.
+// only use default value if some key is empty
+function mergeRomeConfig(
+	config: Partial<RomeConfiguration>,
+	defaultConfig: RomeConfiguration,
+): RomeConfiguration {
+	let interested_keys: Array<keyof RomeConfiguration> = ['treeStyle'];
```
Comment (ematipico) [inline website/playground/src/utils.ts:244]: We already have `persistKeys` which has the same type. We should not repeat code. Also, `const` is enough.

**Review by ematipico — CHANGES_REQUESTED** [review]

File: website/playground/src/utils.ts [hunk 2]
```diff
@@ -17,7 +17,9 @@ export function classNames(
 ): string {
 	return classes.filter(Boolean).join(" ");
 }
-export const persistKeys: Array<keyof PlaygroundState> = ["treeStyle"];
+// Excluding `playgroundState.code` reduce the content to save,
+// Only save prop of `PlaygroundState` that we interested, making `localStorage.setItem` which is file IO a bit faster
```
Comment (ematipico) [inline website/playground/src/utils.ts:21]: This line should have stayed where we use `localStorage.setItem` and you have to add "faster than..." something. I don't know what though... you have to tell us 😛

Comment (IWANABETHATGUY) [reply website/playground/src/utils.ts]: Done

**Review by ematipico — APPROVED** [review]

**Pull request merged**